raft/src/main/java/org/apache/kafka/raft/ReplicatedLog.java

```diff
@@ -27,13 +27,15 @@
      * Write a set of records to the local leader log. These messages will either
      * be written atomically in a single batch or the call will fail and raise an
      * exception.
+     *
+     * @return the last log end offset.
```

Comment:
I don't think this is right. I believe we return the base offset of the first appended record.